### ISSUE TITLE
output diffs after HF→Megatron conversion

### ISSUE THREAD

**Issue by tang-hy** [alibaba/Pai-Megatron-Patch#605]
I used the provided conversion script to transform the Hugging Face Qwen2.5-7B-Instruct checkpoint into Megatron-Core format, and the script reported a successful conversion. However, I observe large numerical differences in the outputs—even though the weights themselves match exactly.

Could you let me know if these numerical differences are expected? Alternatively, could you share a checkpoint that you’ve successfully converted? Thanks!

the diff of outputs：
`layer:0, layer_in, diff: 0, diff>1e-05:[0/35840] diff_max:0.0
layer:0, o_proj_in, diff: 28629, diff>1e-05:[14461/35840] diff_max:0.48046875
layer:0, o_proj_out, diff: 33669, diff>1e-05:[16865/35840] diff_max:0.125
layer:0, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:0, attn_out, diff: 33669, diff>1e-05:[16865/35840] diff_max:0.125
layer:1, layer_in, diff: 33551, diff>1e-05:[16625/35840] diff_max:0.23828125
layer:1, o_proj_in, diff: 33922, diff>1e-05:[16747/35840] diff_max:0.265625
layer:1, o_proj_out, diff: 34192, diff>1e-05:[16985/35840] diff_max:0.1328125
layer:1, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:1, attn_out, diff: 34192, diff>1e-05:[16985/35840] diff_max:0.1328125
layer:2, layer_in, diff: 33148, diff>1e-05:[16485/35840] diff_max:0.296875
layer:2, o_proj_in, diff: 34134, diff>1e-05:[16931/35840] diff_max:0.1953125
layer:2, o_proj_out, diff: 34330, diff>1e-05:[16959/35840] diff_max:0.18359375
layer:2, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:2, attn_out, diff: 34330, diff>1e-05:[16959/35840] diff_max:0.18359375
layer:3, layer_in, diff: 32672, diff>1e-05:[16202/35840] diff_max:0.5625
layer:3, o_proj_in, diff: 33514, diff>1e-05:[16964/35840] diff_max:0.20703125
layer:3, o_proj_out, diff: 33759, diff>1e-05:[16941/35840] diff_max:0.15625
layer:3, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:3, attn_out, diff: 33759, diff>1e-05:[16941/35840] diff_max:0.15625
layer:4, layer_in, diff: 33294, diff>1e-05:[16548/35840] diff_max:8.75
layer:4, o_proj_in, diff: 34003, diff>1e-05:[16748/35840] diff_max:0.14453125
layer:4, o_proj_out, diff: 34086, diff>1e-05:[16841/35840] diff_max:0.1640625
layer:4, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:4, attn_out, diff: 34086, diff>1e-05:[16841/35840] diff_max:0.1640625
layer:5, layer_in, diff: 33426, diff>1e-05:[16634/35840] diff_max:6.5
layer:5, o_proj_in, diff: 33845, diff>1e-05:[16999/35840] diff_max:0.2890625
layer:5, o_proj_out, diff: 33809, diff>1e-05:[16974/35840] diff_max:0.515625
layer:5, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:5, attn_out, diff: 33809, diff>1e-05:[16974/35840] diff_max:0.515625
layer:6, layer_in, diff: 33739, diff>1e-05:[16853/35840] diff_max:4.75
layer:6, o_proj_in, diff: 33888, diff>1e-05:[17033/35840] diff_max:0.2109375
layer:6, o_proj_out, diff: 34030, diff>1e-05:[17038/35840] diff_max:0.21875
layer:6, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:6, attn_out, diff: 34030, diff>1e-05:[17038/35840] diff_max:0.21875
layer:7, layer_in, diff: 33826, diff>1e-05:[16896/35840] diff_max:4.875
layer:7, o_proj_in, diff: 34284, diff>1e-05:[17592/35840] diff_max:0.4140625
layer:7, o_proj_out, diff: 34332, diff>1e-05:[17339/35840] diff_max:0.328125
layer:7, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:7, attn_out, diff: 34332, diff>1e-05:[17339/35840] diff_max:0.328125
layer:8, layer_in, diff: 34048, diff>1e-05:[17054/35840] diff_max:6.125
layer:8, o_proj_in, diff: 34153, diff>1e-05:[17712/35840] diff_max:0.35546875
layer:8, o_proj_out, diff: 34099, diff>1e-05:[17165/35840] diff_max:0.1796875
layer:8, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:8, attn_out, diff: 34099, diff>1e-05:[17165/35840] diff_max:0.1796875
layer:9, layer_in, diff: 34151, diff>1e-05:[17111/35840] diff_max:6.25
layer:9, o_proj_in, diff: 34251, diff>1e-05:[17337/35840] diff_max:0.2490234375
layer:9, o_proj_out, diff: 34369, diff>1e-05:[17322/35840] diff_max:0.65625
layer:9, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:9, attn_out, diff: 34369, diff>1e-05:[17322/35840] diff_max:0.65625
layer:10, layer_in, diff: 34104, diff>1e-05:[17197/35840] diff_max:7.0
layer:10, o_proj_in, diff: 34355, diff>1e-05:[17128/35840] diff_max:0.1875
layer:10, o_proj_out, diff: 34460, diff>1e-05:[17260/35840] diff_max:0.3359375
layer:10, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:10, attn_out, diff: 34460, diff>1e-05:[17260/35840] diff_max:0.3359375
layer:11, layer_in, diff: 34225, diff>1e-05:[17199/35840] diff_max:6.96875
layer:11, o_proj_in, diff: 34148, diff>1e-05:[17208/35840] diff_max:0.16015625
layer:11, o_proj_out, diff: 34208, diff>1e-05:[16893/35840] diff_max:0.3125
layer:11, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:11, attn_out, diff: 34208, diff>1e-05:[16893/35840] diff_max:0.3125
layer:12, layer_in, diff: 34277, diff>1e-05:[17183/35840] diff_max:7.125
layer:12, o_proj_in, diff: 34173, diff>1e-05:[16480/35840] diff_max:0.126953125
layer:12, o_proj_out, diff: 34297, diff>1e-05:[17208/35840] diff_max:0.375
layer:12, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:12, attn_out, diff: 34297, diff>1e-05:[17208/35840] diff_max:0.375
layer:13, layer_in, diff: 34317, diff>1e-05:[17172/35840] diff_max:6.40625
layer:13, o_proj_in, diff: 34243, diff>1e-05:[16780/35840] diff_max:0.1796875
layer:13, o_proj_out, diff: 34382, diff>1e-05:[17123/35840] diff_max:0.25390625
layer:13, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:13, attn_out, diff: 34382, diff>1e-05:[17123/35840] diff_max:0.25390625
layer:14, layer_in, diff: 34262, diff>1e-05:[17093/35840] diff_max:6.65625
layer:14, o_proj_in, diff: 34130, diff>1e-05:[17421/35840] diff_max:0.208984375
layer:14, o_proj_out, diff: 34202, diff>1e-05:[17133/35840] diff_max:0.49609375
layer:14, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:14, attn_out, diff: 34202, diff>1e-05:[17133/35840] diff_max:0.49609375
layer:15, layer_in, diff: 33973, diff>1e-05:[16886/35840] diff_max:6.75
layer:15, o_proj_in, diff: 33955, diff>1e-05:[16577/35840] diff_max:0.171875
layer:15, o_proj_out, diff: 34108, diff>1e-05:[17234/35840] diff_max:0.1953125
layer:15, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:15, attn_out, diff: 34108, diff>1e-05:[17234/35840] diff_max:0.1953125
layer:16, layer_in, diff: 34119, diff>1e-05:[16945/35840] diff_max:7.0
layer:16, o_proj_in, diff: 34152, diff>1e-05:[16647/35840] diff_max:0.15625
layer:16, o_proj_out, diff: 34277, diff>1e-05:[17028/35840] diff_max:0.40625
layer:16, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:16, attn_out, diff: 34277, diff>1e-05:[17028/35840] diff_max:0.40625
layer:17, layer_in, diff: 34189, diff>1e-05:[16998/35840] diff_max:7.4375
layer:17, o_proj_in, diff: 34153, diff>1e-05:[17154/35840] diff_max:0.15625
layer:17, o_proj_out, diff: 34175, diff>1e-05:[17094/35840] diff_max:0.28515625
layer:17, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:17, attn_out, diff: 34175, diff>1e-05:[17094/35840] diff_max:0.28515625
layer:18, layer_in, diff: 34247, diff>1e-05:[17055/35840] diff_max:7.21875
layer:18, o_proj_in, diff: 34191, diff>1e-05:[17176/35840] diff_max:0.197265625
layer:18, o_proj_out, diff: 34386, diff>1e-05:[16969/35840] diff_max:0.375
layer:18, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:18, attn_out, diff: 34386, diff>1e-05:[16969/35840] diff_max:0.375
layer:19, layer_in, diff: 34269, diff>1e-05:[17125/35840] diff_max:7.0
layer:19, o_proj_in, diff: 34285, diff>1e-05:[17851/35840] diff_max:0.21484375
layer:19, o_proj_out, diff: 34413, diff>1e-05:[17443/35840] diff_max:0.9140625
layer:19, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:19, attn_out, diff: 34413, diff>1e-05:[17443/35840] diff_max:0.9140625
layer:20, layer_in, diff: 34231, diff>1e-05:[17124/35840] diff_max:6.46875
layer:20, o_proj_in, diff: 34333, diff>1e-05:[17078/35840] diff_max:0.25390625
layer:20, o_proj_out, diff: 34437, diff>1e-05:[16917/35840] diff_max:0.875
layer:20, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:20, attn_out, diff: 34437, diff>1e-05:[16917/35840] diff_max:0.875
layer:21, layer_in, diff: 34272, diff>1e-05:[17110/35840] diff_max:5.25
layer:21, o_proj_in, diff: 34385, diff>1e-05:[16800/35840] diff_max:0.3359375
layer:21, o_proj_out, diff: 34450, diff>1e-05:[17022/35840] diff_max:0.6953125
layer:21, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:21, attn_out, diff: 34450, diff>1e-05:[17022/35840] diff_max:0.6953125
layer:22, layer_in, diff: 34167, diff>1e-05:[17108/35840] diff_max:5.59375
layer:22, o_proj_in, diff: 34308, diff>1e-05:[16958/35840] diff_max:1.234375
layer:22, o_proj_out, diff: 34478, diff>1e-05:[17165/35840] diff_max:0.921875
layer:22, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:22, attn_out, diff: 34478, diff>1e-05:[17165/35840] diff_max:0.921875
layer:23, layer_in, diff: 34356, diff>1e-05:[17149/35840] diff_max:5.125
layer:23, o_proj_in, diff: 34360, diff>1e-05:[16915/35840] diff_max:0.953125
layer:23, o_proj_out, diff: 34350, diff>1e-05:[17238/35840] diff_max:1.203125
layer:23, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:23, attn_out, diff: 34350, diff>1e-05:[17238/35840] diff_max:1.203125
layer:24, layer_in, diff: 34202, diff>1e-05:[17080/35840] diff_max:5.1875
layer:24, o_proj_in, diff: 34271, diff>1e-05:[16459/35840] diff_max:0.5625
layer:24, o_proj_out, diff: 34359, diff>1e-05:[17231/35840] diff_max:0.875
layer:24, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:24, attn_out, diff: 34359, diff>1e-05:[17231/35840] diff_max:0.875
layer:25, layer_in, diff: 34346, diff>1e-05:[17201/35840] diff_max:5.0625
layer:25, o_proj_in, diff: 34248, diff>1e-05:[17452/35840] diff_max:0.6875
layer:25, o_proj_out, diff: 34344, diff>1e-05:[16793/35840] diff_max:1.40625
layer:25, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:25, attn_out, diff: 34344, diff>1e-05:[16793/35840] diff_max:1.40625
layer:26, layer_in, diff: 34320, diff>1e-05:[17120/35840] diff_max:6.0
layer:26, o_proj_in, diff: 34384, diff>1e-05:[17123/35840] diff_max:0.91015625
layer:26, o_proj_out, diff: 34412, diff>1e-05:[17244/35840] diff_max:2.75
layer:26, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:26, attn_out, diff: 34412, diff>1e-05:[17244/35840] diff_max:2.75
layer:27, layer_in, diff: 34351, diff>1e-05:[17071/35840] diff_max:13.0
layer:27, o_proj_in, diff: 34366, diff>1e-05:[17132/35840] diff_max:5.3125
layer:27, o_proj_out, diff: 34589, diff>1e-05:[17224/35840] diff_max:11.0
layer:27, o_proj_out_weight, diff: 0, diff>1e-05:[0/12845056] diff_max:0.0
layer:27, attn_out, diff: 34589, diff>1e-05:[17224/35840] diff_max:11.0
layer:27, lmhead, diff: 1452168, diff>1e-05:[819671/1520640] diff_max:1.3125
layer:27, lmhead_weight, diff: 0, diff>1e-05:[0/544997376] diff_max:0.0
layer:27, lmhead_token, diff: 3, diff>1e-05:[2/10] diff_max:72779
logits: 1452168, diff>1e-05:[819671/1520640] diff_max:1.3125`

**Comment by lostkevin** [comment]
Thanks for reporting! This accumulated numerical difference is only for debugging and can be ignored for most cases. We'll remove the code in the future to avoid confusing users.
